content/200-orm/200-prisma-client/150-using-raw-sql/100-typedsql.mdx

```diff
@@ -0,0 +1,76 @@
+---
+title: 'TypedSQL'
+metaTitle: 'Writing Type-safe SQL with TypedSQL and Prisma Client'
+metaDescription: 'Learn how to use TypedSQL to write fully type-safe SQL queries that are compatible with any SQL console and Prisma Client.'
+---
+
+## Getting started with TypedSQL
+
+To start using TypedSQL in your Prisma project, follow these steps:
+
+1. Ensure you have `@prisma/client` and `prisma` installed and updated to at least version `5.19.0`.
+
+2. Create a `sql` directory inside your `prisma` directory. This is where you'll write your SQL queries.
+
+   ```terminal
+   cd prisma
+   mkdir sql
+   ```
+
+3. Write your SQL queries in `.sql` files within the `sql` directory. For example:
+
+   ```sql title="prisma/sql/getUsersWithPosts.sql"
+   SELECT u.id, u.name, COUNT(p.id) as postCount
+   FROM users u
+   LEFT JOIN posts p ON u.id = p.authorId
+   GROUP BY u.id, u.name
+   ```
+
+4. Generate Prisma Client with the `sql` flag to ensure TypeScript functions and types for your SQL queries are created:
```

Comment:
Probably would be worth noting that any pending migrations if exist should be ran before `generate`